scipy/special/tests/test_basic.py

```diff
@@ -1845,6 +1845,37 @@ def test_fresnel(self):
         frs = array(special.fresnel(.5))
         assert_array_almost_equal(frs,array([0.064732432859999287, 0.49234422587144644]),8)
 
+    def test_fresnel_negative_half_annulus(self):
+        # https://github.com/scipy/scipy/issues/12309
+        frs = array(special.fresnel(-2.0 + 0.1j))
+        assert_array_almost_equal(
+            frs,
+            array([
```

Comment:
Add a comment explaining where these values came from so that a reader can reproduce them.

For bonus points; add something to `special/_precompute` that does say a naive computation via quadrature say and use those values in the tests so that we have a programmatic reproducer.